impl/src/main/java/impl/QuestionResourceImpl.java

```diff
@@ -111,6 +120,29 @@ public QuestionResourceImpl(QuestionDao questionDao, QuestionVoteDao questionVot
         return handleError(questionDao.getRecentlyAcceptedQuestions(options), FAILED_TO_GET_RECENTLY_ACCEPTED_QUESTIONS);
     }
 
+    private Observable<Tag> storeTagsMerging(Set<String> requestedLabels) {
+        return tagDao
+            .getTagsByLabels(Lists.newArrayList(requestedLabels))
+            .collect(HashSet<Tag>::new, Set::add)
+            .switchIfEmpty(just(new HashSet<>()))
+            .concatMap(foundTags -> {
+                Set<String> foundLabels = foundTags
+                    .stream()
+                    .map(Tag::getLabel)
+                    .collect(Collectors.toSet());
+
+                List<Observable<Tag>> createTagOperations = Sets.difference(requestedLabels, foundLabels)
+                    .stream()
+                    .map(missingLabel -> tagDao.createTag(missingLabel).map(GeneratedKey::getKey))
```

Comment:
Should we handle error for when tag already exists? I guess that could happen if tags are added around the same time.

Reply:
I'll  take a look at adding ON CONFLICT DO NOTHING to TagDao#createTag. I may need some help writing a test for that.

Reply:
Tags are now added to the table in a way that wont conflict with existing rows.